File: packages/common/erp-common-ms-3-js/src/modules/v1/country/country.controller.ts
import { ICountry, ICreateCountry, IUpdateCountry, IDeleteCountry, ISearchCountry } from '@gmahechas/erp-common-js';
import { Country } from './country.mongo';

export const createOneCountry = async (data: ICreateCountry): Promise<ICountry> => {
	let result = await Country.create(data);
	return result;
};

export const updateOneCountry = async (data: IUpdateCountry): Promise<ICountry | null> => {
	let result = await Country.findOneAndUpdate({ countryId: data.countryId }, data);
	// result = Object.assign(result, data);
	return result;
};

export const deleteOneCountry = async (data: IDeleteCountry): Promise<ICountry | null> => {
	let result = await Country.findOneAndDelete(data);
	return result;
};

export const searchOneCountry = async (data: Partial<ISearchCountry>): Promise<ICountry | null> => {
	let result = await Country.findOne(data);
	return result;
};

export const searchManyCountry = async (data: Partial<ISearchCountry>[]): Promise<ICountry[]> => {
	const search = data.length === 0 ? [{}] : data;
	let result = await Country.find({ $or: search });
	return result;
};

File: packages/common/erp-common-ms-3-py/gmahechas_erp_common_ms_3_py/modules/v1/country/country_controller.py
from .country_model import CountryModel


class CountryController:

    def __init__(self):
        self.country_model = CountryModel()

    def create_one_country(self, data):
        result = self.country_model.create_one(data)
        return result

    def update_one_country(self, data):
        result = self.country_model.update_one(data)
        return result

    def delete_one_country(self, data):
        result = self.country_model.delete_one(data)
        return result

    def search_one_country(self, data):
        result = self.country_model.search_one(data)
        return result

    def search_many_country(self, data):
        result = self.country_model.search_many(data)
        return result
